metamist/parser/generic_metadata_parser.py

```diff
@@ -553,7 +608,7 @@ async def get_sequencing_group_meta(
 
         return meta
 
-    async def get_assays_from_group(
+    async def get_assays_from_group(  # pylint: disable=too-many-branches
```

Comment:
I wonder if all the read and checksum can be extracted in a separate function, or maybe even that section if reads_type == 'cram' as a separate fun. 
Just making it a bit easier to read.
As an outsider looks like there is a lot of going on there.

Reply:
Cheers, I split this out into a function `get_read_and_ref_files_and_checksums` which parses the read filenames, checksums, and reference assemblies; and another separate new function `parse_cram_assays` which parses any CRAM assays and links the reference assemblies to the secondary files. No code changes, just splitting the large function into a few smaller ones.